xrviz/dashboard.py

```diff
@@ -31,6 +33,7 @@ def __init__(self, data):
         if not isinstance(data, xr.core.dataarray.DataWithCoords):
             raise ValueError("Input should be an xarray data object, not %s" % type(data))
         self.set_data(data)
+        cartopy_geoviews_installed()
```

Comment:
It is odd to have this run every time.
Better, I think, would be to have a compatibility module which we import all the optional modules from, so that the checks happen only once
```
from .compatability import ccrs, gv, g, has_cartopy
```

and in the module:
```
try:
    from cartopy import crs as ccrs
    import geoviews as gv
    import geoviews.feature as gf
    has_cartopy = True
except ImportError:
    logging.debug('...')
```

That way you only call it once, and you can reuse this in other places. The downside would be that this happens on import of xrviz *unless* we contrive a way to do the import and keep hold of the references during `__init__`. I find the simple method above is much easier to understand, so probably worth any small import time cost.

Reply:
This method works perfect and solves earlier problems. Thanks!